201-representing-CNAB-in-OCI.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Representing CNAB bundles in OCI Registries
+weight: 201
+---
+
+# Representing CNAB bundles in OCI Registries
+
+This section describes how the bundle file and the information represented in the bundle MUST be represented when distributing bundles using OCI registries.
+
+A CNAB is made up of a descriptor file (`bundle.json`), one or more invocation images, and a list of zero or more component images. Each of these objects can be individually packaged into an [OCI manifest][oci-manifest] or [OCI index][oci-index]. An OCI index can then be used to combine these all into a single object.
+
+Runtimes and registries MAY choose to _relocate_ the images and invocation images referenced in the bundle to the registry where the bundle is pushed, but in its simplest form, a CNAB bundle MAY be represented in an OCI registry by the canonical `bundle.json` file descriptor, referenced in an OCI index (or manifest list).
+
+![](https://i.imgur.com/PfTcKOm.png)
+
+### `bundle.json` manifest
+
+The `bundle.json` MUST be serialized as canonical JSON and MUST be stored in the registry as a blob. This blob MUST be referenced by its digest in an OCI manifest. The manifest media type SHOULD be `application/vnd.cnab.bundle.config.v1+json` but MAY be a standard container image config type (`application/vnd.oci.image.config.v1+json`) if the target registry does not support the CNAB media type.
```

Comment:
Nit/question: is `application/vnd.cnab.bundle.config.v1+json` redundant? We've been using `application/vnd.cnab.config.v1+json` internally. I can see the former being nice if we add more cnab-scoped artifact types in the future though.

Reply:
I think at this point we should follow the [guidance in the OCI Artifacts repository](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md#registering-unique-types-with-iana).

Reply:
Opened #381 to track this.